lib/Sema/CSDiag.cpp

```diff
@@ -2617,7 +2618,7 @@ diagnoseTypeMemberOnInstanceLookup(Type baseObjTy,
         // Fetch any declaration to check if the name is '~='
         ValueDecl *decl0 = overloadedFn->getDecls()[0];
 
-        if (decl0->getName() == decl0->getASTContext().Id_MatchOperator) {
+        if (decl0->getBaseName() == decl0->getASTContext().Id_MatchOperator) {
```

Comment:
My scheme was actually `getBaseName` everywhere and `getFullName` just if comparing against `self`. The idea behind it was that I don't care about the operator's argument labels in this context. In the `self` case, I wouldn't want to match some variable that has a composite name (or can guarantee that `self` never has a composite name).